lib/format-utils.js

```diff
@@ -100,11 +106,40 @@ exports.chooseFormat = (formats, options) => {
     return options.format;
   }
 
+  let format;
+  if (formats.some(fmt => fmt.isHLS)) {
+    const testAudioQuality = options.quality &&
+      typeof options.quality === 'string' &&
+      /(lowest|highest)audio/.test(options.quality);
+    const testAudioFilter = options.filter &&
+      typeof options.filter === 'string' &&
+      /audio(only)?/.test(options.filter);
+
+    if (testAudioQuality || testAudioFilter) {
+      formats = formats.filter(fmt => fmt.isHLS);
+      if (testAudioQuality) {
+        switch (options.quality) {
+          case 'highestaudio':
+            formats.sort(sortFormatsByHighAudioAndLowVideo);
```

Comment:
is this not also needed for 'lowestaudio' and therefor all 3 cases?

Reply:
There's no need.
I'm already filtering out all formats that aren't HLS from the array, so the lowest audio is automatically in the end.

`sortFormatsByHighAudioAndLowVideo` exists because, typically, some resolutions have the same audio bitrate.
p.e. it makes it so that if both `720p` and `1080p`exist (these typically have the same audio bitrate), it'll sort `720p` to the top of the array and choose it in order to reduce bandwidth use for the user.